geode-gfsh/src/main/java/org/apache/geode/management/internal/cli/functions/WanCopyRegionFunction.java

```diff
@@ -0,0 +1,557 @@
+    } catch (EntryDestroyedException e) {
+      return null;
+    }
+    if (entry instanceof NonTXEntry) {
+      event.setVersionTag(((NonTXEntry) entry).getRegionEntry().getVersionStamp().asVersionTag());
+    } else {
+      event.setVersionTag(((EntrySnapshot) entry).getVersionTag());
+    }
+    event.setNewEventId(cache.getInternalDistributedSystem());
+    return event;
+  }
+
+  static class ConnectionState {
+    private volatile Connection connection = null;
+    private volatile PoolImpl senderPool = null;
+
+    public Connection getConnection() {
+      return connection;
+    }
+
+    public PoolImpl getSenderPool() {
+      return senderPool;
+    }
+
+    public Optional<CliFunctionResult> connectIfNeeded(FunctionContext<Object[]> context,
+        GatewaySender sender) {
+      if (senderPool == null) {
+        senderPool = ((AbstractGatewaySender) sender).getProxy();
+        if (senderPool == null) {
+          return Optional.of(new CliFunctionResult(context.getMemberName(),
+              CliFunctionResult.StatusState.ERROR,
+              CliStrings.WAN_COPY_REGION__MSG__NO__CONNECTION__POOL));
+        }
+        connection = senderPool.acquireConnection();
+        if (connection.getWanSiteVersion() < KnownVersion.GEODE_1_15_0.ordinal()) {
+          return Optional.of(new CliFunctionResult(context.getMemberName(),
+              CliFunctionResult.StatusState.ERROR,
+              CliStrings.WAN_COPY_REGION__MSG__COMMAND__NOT__SUPPORTED__AT__REMOTE__SITE));
+        }
+      }
+      return Optional.empty();
+    }
+
+    public Optional<CliFunctionResult> reconnect(FunctionContext<Object[]> context, int retries,
+        int copiedEntries, Exception e) {
+      ((PooledConnection) connection).setShouldDestroy();
+      senderPool.returnConnection(connection);
+      connection = null;
+      if (retries++ >= MAX_BATCH_SEND_RETRIES) {
+        return Optional.of(new CliFunctionResult(context.getMemberName(),
+            CliFunctionResult.StatusState.ERROR,
+            CliStrings.format(
+                CliStrings.WAN_COPY_REGION__MSG__ERROR__AFTER__HAVING__COPIED,
+                "Connection error", copiedEntries)));
+      }
+      logger.error("Exception {} in sendBatch. Retrying", e.getClass().getName());
+      try {
+        connection = senderPool.acquireConnection();
+      } catch (NoAvailableServersException | AllConnectionsInUseException e1) {
+        return Optional.of(new CliFunctionResult(context.getMemberName(),
+            CliFunctionResult.StatusState.ERROR,
+            CliStrings.format(
+                CliStrings.WAN_COPY_REGION__MSG__NO__CONNECTION,
+                copiedEntries)));
+      }
+      return Optional.empty();
+    }
+
+    void close() {
+      if (senderPool != null && connection != null) {
+        ((PooledConnection) connection).setShouldDestroy();
+        senderPool.returnConnection(connection);
+      }
```

Comment:
When the sending members finish copying the region, the receiving members log warnings like this for each connection:
```
[warn 2021/06/24 14:00:53.172 PDT ny-1 <ServerConnection on port 5025 Thread 2> tid=0x69] Server connection from [identity(10.166.144.101(ln-1:76734)<v1>:41003,connection=2; port=64429]: connection disconnect detected by EOF.
```
The `setShouldDestroy` and `returnConnection` call here cause the socket to be closed which causes that warning.

There might need to be another parameter or API that closes the connection (which closes the socket nicely) rather than destroying it.

I hacked `returnConnection` to see if it would get rid of the warn message (it does) but its not a valid fix:
```
private void returnConnection(Connection connection, boolean accessed, boolean addLast) {
  ...
  if (pooledConn.shouldDestroy()) {
    try {pooledConn.close(false);} catch (Exception e) {}
    destroyConnection(pooledConn);
  } ...
}
```

Reply:
I've changed the call to `setShouldDestroy()` to close(). This way the connection is closed gracefully and we get rid of the log warnings in the receiving members.